### PR TITLE
added wordtokenizer

### PR DESCRIPTION
**Pull request by DecayingSeart**
Wordtokenizer added. Still open for test before merging.

### CODE REVIEW

File: src/conf/main.yaml
Comment (adambarla) [inline src/conf/main.yaml:1]: this represents the default. please revert the changes so character stays as the default

**Review by adambarla — CHANGES_REQUESTED** [review]

**Comment by DecayingSeart** [comment]
Now it can be run properly (I'm waiting for its results). Vocab size is now 10000 (used unk).

**Comment by adambarla** [comment]
could you please remove the debug prints? also do we need to keep the old version of `word.py`?

File: src/conf/main.yaml
Comment (adambarla) [inline src/conf/main.yaml:1]: please don't commit your local changes in the configs or don't change them at all and just use `device="cuda:3"` in the training command

File: src/tokenizer/word_old.py
Comment (adambarla) [inline src/tokenizer/word_old.py:1]: is this file needed?

File: src/tokenizer/word.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+            encoded = encoded[: max_length - (2 if add_special_tokens else 0)]
+        if add_special_tokens:
+            encoded = [self._stoi["<s>"]] + encoded + [self._stoi["</s>"]]
+        if padding == "max_length" and max_length is not None:
+            encoded += [self.pad_token_id] * (max_length - len(encoded))
+        return encoded
+
+    def decode(self, x):
+        decoded_text = " ".join([self._itos.get(w, "<unk>") for w in x])
+        print(f"Decoded text: {decoded_text}")
```
Comment (adambarla) [inline src/tokenizer/word.py]: is this print necessary?

File: src/tokenizer/word.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        self.min_freq = min_freq
+        self.vocab = self._create_vocab(dataset)
+        self._stoi = {w: i for i, w in enumerate(self.vocab)}
+        self._itos = {i: w for i, w in enumerate(self.vocab)}
+        self.vocab_size = len(self.vocab)
+        self.pad_token_id = self._stoi["<pad>"]
+        self.unk_token_id = self._stoi["<unk>"]
+        print(f"Vocabulary size: {self.vocab_size}")
+        print(f"Pad token ID: {self.pad_token_id}")
+        print(f"Unknown token ID: {self.unk_token_id}")
```
Comment (adambarla) [inline src/tokenizer/word.py]: are these prints necessary?

File: src/training.py [hunk 3]
```diff
@@ -105,13 +105,18 @@ def main(cfg):
         log_with="wandb",
         # logging_dir="logs" # unexpected argument?
     )
-    device = get_device(cfg)
+    
+    device = get_device(cfg) #torch.device("cuda:1")
```
Comment (adambarla) [inline src/training.py]: remove this comment please

File: src/training.py [hunk 4]
```diff
@@ -105,13 +105,18 @@ def main(cfg):
...
     dataset = get_dataset(cfg)
     tokenizer = hydra.utils.instantiate(cfg.tokenizer, dataset=dataset)
     cfg.src_vocab_size = tokenizer.vocab_size
     cfg.tgt_vocab_size = tokenizer.vocab_size
     print(f"Tokenizer:\n{tokenizer}")
 
+    #added by me
```
Comment (adambarla) [inline src/training.py]: and this one

**Review by adambarla — CHANGES_REQUESTED** [review]

File: src/training.py (hunk 4 above)
Comment (DecayingSeart) [reply src/training.py]: Fixed

**Review by adambarla — APPROVED** [review]

**Pull request merged**